README.md

```diff
@@ -351,7 +354,207 @@ The last step will launch Jenkins tests. If these pass:
+FV3core model using Pythong wrapped fortran for code not ported to GT4py at the moment.
+
+The namelist and grid are global variables defined in fv3core/_config.py The namelist is
+'flattened' so that the grouping name of the option is not required to access the data
+(we may want to change this).
+
+The grid variables are mostly 2d variables and are 'global' to the model thread per mpi rank. The
+grid object also contains domain and layout information relevant to the current
+rank being operated on.
+
+Utility functions in `utils/` include:
+  - `gt4py_utils.py`:
+    - default gt4py and model settings
+    - methods for generating gt4py storages
+    - methods for using numpy and cupy arrays in python functions that have not been
+      put into GT4py
+    - methods for handling complex patterns that did not immediately map to gt4py,
+      and will mostly be removed with future refactors (e.g. k_split_run)
+    - some general model math computations (e.g. great_circle_dist), that will
+      eventually be put into gt4py with a future refactor
+  - `grid.py`:
+    - A Grid class definition that provides information about the grid layout,
+      current tile informationm access to grid variables used globally, and
+      convenience methods related to tile indexing, origins and domains commonly used
+    - A grid is defined for each MPI rank (minimum 6 ranks, 1 for each tile face of
+      the cubed sphere grid represnting the whole Earth)
+    - Also provides functionality for generating a Quantity object (used to interface
+      with the fv3gfs-wrapper, that allows us to run the full model, not just the
+      dynamical core)
+  - `corners`: port of corner calculations, initially direct Python calculations, being
+    replaced with GT4py gtscript functions as the GT4py regions feature is implemented
+  - `mpi.py`: a wrapper for importing mpi4py when available
+  - `global_constants.py`: constants for use throughout the model
+  - `typing.py`: Clean names for common types we use in the model. This is new and
+    hasn't been adopted throughout the model yet, but will eventually be our
+    standard. A shorthand 'sd' has been used in the intial version.
+
+The `tests/` directory currently includes a framework for translating fields serialized (using
+Serialbox from GridTools) from a Fortran run into gt4py storages that can be inputs to
+fv3core unit computations, and compares the results of the ported code to serialized
+data following a unit computation.
+
+The `docker/` directory provides Dockerfiles for building a repeatable environment in which
+to run the core
+
+The `external/` directory is for submoduled repos that provide essential functionality
+
+The build system uses Makefiles following the convention of other repos within VulcanClimateModeling.
+
+## Model Interface
+The top level functions fv_dynamics and fv_sugridz can currenty only be run in parallel
```

Comment:
Could you please single-line-ify the lines in the rest of the document? If you really don't feel like it, you can leave as-is.